core/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -298,8 +304,24 @@ public GetResult extractGetResult(final UpdateRequest request, String concreteIn
             }
         }
 
+        BytesReference sourceFilteredAsBytes = sourceAsBytes;
+        if (request.fetchSource() != null) {
```

Comment:
Maybe just handle the boolean for this PR, but we should think about removing it....